ql.go

```diff
@@ -516,6 +528,33 @@ func (r *whereRset) planUnaryOp(x *unaryOperation) (plan, error) {
 }
 
 func (r *whereRset) plan(ctx *execCtx) (plan, error) {
+	if r.sel != nil {
+		var exists bool
+		p, err := r.sel.plan(ctx)
+		if err != nil {
+			return nil, err
+		}
+		err = p.do(ctx, func(i interface{}, data []interface{}) (bool, error) {
+			if len(data) > 0 {
```

Comment:
Is it possible that the first result set might be zero but the second might not be zero?

I mean 

Instead of 
```go
	err = p.do(ctx, func(i interface{}, data []interface{}) (bool, error) {
			if len(data) > 0 {
				exists = true
			}
			return true, nil
		})
```
To have
```go
		err = p.do(ctx, func(i interface{}, data []interface{}) (bool, error) {
			if len(data) > 0 {
				exists = true
				return false, nil
			}
			return true, nil
		})
```

On the second snippet we are asking for more result sets to check if there is any data matched